File: java-web-performance-springmvc/src/main/java/com/binghe/MemberController.java
package com.binghe;

import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class MemberController {

    private final MemberService memberService;

    @GetMapping("/member/{id}")
    public ResponseEntity<Member> findById(@PathVariable Long id) {
        return ResponseEntity.ok().body(memberService.findById(id));
    }
}

File: kotlin-web-performance-webflux-coroutine/src/main/kotlin/com/binghe/MemberController.kt
package com.binghe

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController

@RestController
class MemberController(private val memberService: MemberService) {

    @GetMapping("/member/{id}")
    suspend fun findById(@PathVariable id: Long): Member? =
        memberService.findById(id)
}
